### PR TITLE
Datasets overview docs

### PR DESCRIPTION
**Pull request by RdoubleA**
#### Context
What is the purpose of this PR? Is it to
- [ ] add a new feature
- [ ] fix a bug
- [ ] update tests and/or documentation
- [ ] other (please add here)

Please link to any issues this PR addresses.

#### Changelog
What are the changes made in this PR?
*

#### Test plan
Please make sure to do each of the following if applicable to your PR. If you're unsure about any one of these just ask and we will happily help. We also have a [contributing page](https://github.com/pytorch/torchtune/blob/main/CONTRIBUTING.md) for some guidance on contributing.

- [ ] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [ ] add [unit tests](https://github.com/pytorch/torchtune/tree/main/tests/torchtune) for any new functionality
- [ ] update [docstrings](https://github.com/pytorch/torchtune/tree/main/docs/source) for any new or updated methods or classes
- [ ] run unit tests via `pytest tests`
- [ ] run recipe tests via `pytest tests -m integration_test`
- [ ] manually run any new or modified recipes with sufficient proof of correctness
- [ ] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

#### UX
If your function changed a public API, please add a dummy example of what the user experience will look like when calling it.
Here is a [docstring example](https://github.com/pytorch/torchtune/blob/6a7951f1cdd0b56a9746ef5935106989415f50e3/torchtune/modules/vision_transformer.py#L285)
and a [tutorial example](https://pytorch.org/torchtune/main/tutorials/qat_finetune.html#applying-qat-to-llama3-models)

- [ ] I did not change any public API
- [ ] I have added an example to docs or docstrings

### CODE REVIEW

File: docs/source/basics/datasets_overview.rst [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+Data pipeline
+-------------
+.. image:: /_static/img/torchtune_datasets.png
+
+From raw data samples to the model inputs in the training recipe, all torchtune datasets follow
+the same pipeline:
+
+1. Raw data is queried one sample at a time from a Hugging Face dataset, local file, or remote file
+2. :ref:`message_transform_usage_label` convert the raw sample which can take any format into a list of torchtune
+   :ref:`message_usage_label`. Images are contained in the message object they are associated with.
```
Comment (pbontrager) [inline docs/source/basics/datasets_overview.rst:32]: Maybe instead of "Images" here say "Media"

**Review by pbontrager — APPROVED** [review]
Looks great! Approving so you can land after you fix the docs build.
